### PR TITLE
feat(agent-factory): M1.e narrow first-pass review workflow

### PR DESCRIPTION
**Pull request by lukeelmers**
## M1.e

Narrow first-pass review for `kibana-agent-review`: agent reviews agent-authored PRs from diff, linked issue/spec when present, and CI signals; applies mechanical fixes via a single safe push to the PR branch when appropriate; posts one structured review comment (scope, quality, conventions, tests).

### CODE REVIEW

File: .github/workflows/kibana-agent-review.md [hunk 1]
```diff
@@ -45,15 +45,92 @@ safe-outputs:
...
     target: "*"
     hide-older-comments: true
 
 strict: true
-timeout-minutes: 20
+timeout-minutes: 30
 ---
 
-# Kibana Agent — review (POC stub)
+# Kibana Agent — review (M1.e narrow first-pass)
```
Comment (lukeelmers) [inline .github/workflows/kibana-agent-review.md]: remove mention of M1.e

File: .github/workflows/kibana-agent-review.md [hunk 2]
```diff
@@ -45,15 +45,92 @@ safe-outputs:
...
+
+## 1. Identity and scope
+
+You are **kibana-agent**, reviewing a pull request in the Kibana monorepo.
+
+This is **CR-A**: review of agent-authored PRs. The goal is to catch issues before a human reviewer sees the PR.
+
+You produce **one synthesized review comment** on the PR. Your internal reasoning is not visible to users.
+
+Structure findings around four concerns (this is a **narrow first-pass**; richer persona-based rubrics will extend this list in M2.b without changing this workflow’s shape):
```
Comment (lukeelmers) [inline .github/workflows/kibana-agent-review.md]: remove mention of M2.b

File: .github/workflows/kibana-agent-review.md [hunk 3]
```diff
@@ -45,15 +45,92 @@ safe-outputs:
...
+
+1. **Scope alignment**
+2. **Code quality**
+3. **Kibana conventions**
+4. **Test coverage**
+
+## 2. Input gathering
+
+- Read the PR diff using the GitHub MCP tools (`get_pull_request_diff`, `get_pull_request_files`).
```
Comment (lukeelmers) [inline .github/workflows/kibana-agent-review.md]: isn't it better to use gh cli?

File: .github/workflows/kibana-agent-review.md [hunk 4]
```diff
@@ -45,15 +45,92 @@ safe-outputs:
...
+  - Check CI status via `get_pull_request_status` or workflow run / check APIs you have access to.
+  - Review lint or type-check output in check logs when present.
+
+## 3. Review criteria (narrow first-pass)
+
+Evaluate the diff and context against:
+
+1. **Scope alignment** — Does the diff match what the spec requires? Missing work? Scope creep?
+2. **Code quality** — Obvious bugs, error-handling gaps, naming, dead code, unnecessary complexity.
+3. **Kibana conventions** — File placement, import patterns, TypeScript usage, test layout consistent with this monorepo.
```
Comment (lukeelmers) [inline .github/workflows/kibana-agent-review.md]: specifically we want to be consistent with the code _in this area of the monorepo_. different areas have different conventions. when touching code in a particular plugin or package, it should feel like it belongs with what's around it.

File: .github/workflows/kibana-agent-review.md [hunk 5]
```diff
@@ -45,15 +45,92 @@ safe-outputs:
...
+4. **Test coverage** — Are there tests? Do they cover key behaviors and important edge cases?
```
Comment (lukeelmers) [inline .github/workflows/kibana-agent-review.md]: also we want to make sure we're generally following the testing pyramid (more unit tests, fewer e2e tests). for e2e tests the preferred test runner is scout - FTR (functional test runner) is legacy. the kibana codebase has skills to help with authoring scout tests that can be used if needed.

File: .github/workflows/kibana-agent-review.md [hunk 6]
```diff
@@ -45,15 +45,92 @@ safe-outputs:
...
+
+Ground the review in the **diff**, the **approved spec** (when available), and **deterministic signals** (CI, logs).
+
+Do **not** seek or use internal chain-of-thought, planning artifacts, or hidden rationale from an execution workflow. You are an **independent** reviewer.
+
+## 7. Error handling and skip conditions
 
-Placeholder for automated review comments and follow-on dispatches.
+- If the PR has **no diff** or the diff **cannot be read**, use **`report_incomplete`**.
+- If the PR is **clearly not agent-authored** (e.g. no spec reference and no agent branch pattern you recognize), post a **brief** comment stating that you are skipping because this does not appear to be an agent-authored PR, then stop.
```
Comment (lukeelmers) [inline .github/workflows/kibana-agent-review.md]: no - the reviewer should still run even if it isn't an agent-authored PR. and if no agent-authored spec can be found, just do a more basic check for code correctness

File: .github/workflows/kibana-agent-review.md [hunk 7]
```diff
@@ -1,6 +1,6 @@
 ---
 name: Kibana Agent Review
-description: POC stub — multi-perspective PR review (scaffold only).
+description: Narrow first-pass PR review for agent-authored changes (M1.e); rubric scaffold for later persona depth (M2.b).
```
Comment (lukeelmers) [inline .github/workflows/kibana-agent-review.md]: remove mention of phases.
